eureka2-client/src/main/java/com/netflix/eureka2/client/interest/EurekaInterestClientImpl.java

```diff
@@ -71,54 +57,8 @@ public EurekaInterestClientImpl(SourcedEurekaRegistry<InstanceInfo> registry,
 
         this.retryableConnection = retryableConnectionFactory.singleOpConnection(opStream, executeOnChannel);
 
-        // subscribe to the base interest channels to do cleanup on every channel refresh.
-        retryableConnection.getChannelObservable()
```

Comment:
The original intent for having these in the constructor was to ensure that they are not multi-subscribed to, and also that they are always eagerly subscribed to. Moving these to methods in the abstract client allows breakage of the above assumptions for future implementors. From what I can see the two implementing clients always call
```java
registryEvictionSubscribe(retryableConnection);
lifecycleSubscribe(retryableConnection);
````
anyway in the constructor, so can we just move this block to the constructor for the new Abstract class instead of making them methods?